msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/wpj/AbstractWpjTest.java

```diff
@@ -0,0 +1,169 @@
+import com.microsoft.identity.client.ui.automation.BuildConfig;
+import com.microsoft.identity.client.ui.automation.IBrokerHostTest;
+import com.microsoft.identity.client.ui.automation.IBrokerTest;
+import com.microsoft.identity.client.ui.automation.annotations.SupportedBrokers;
+import com.microsoft.identity.client.ui.automation.broker.BrokerCompanyPortal;
+import com.microsoft.identity.client.ui.automation.broker.BrokerHost;
+import com.microsoft.identity.client.ui.automation.broker.ITestBroker;
+import com.microsoft.identity.client.ui.automation.installer.IAppInstaller;
+import com.microsoft.identity.client.ui.automation.installer.LocalApkInstaller;
+import com.microsoft.identity.client.ui.automation.installer.PlayStore;
+import com.microsoft.identity.client.ui.automation.rules.DevicePinSetupRule;
+import com.microsoft.identity.client.ui.automation.rules.RulesHelper;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserHelper;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+import org.junit.Rule;
+import org.junit.rules.RuleChain;
+
+import java.util.Arrays;
+
+/**
+ * An MSAL test model that would leverage an {@link ITestBroker} and {@link IBrokerHostTest} installed on the device.
+ */
+public abstract class AbstractWpjTest extends AbstractMsalUiTest implements IBrokerTest, IBrokerHostTest {
+
+    protected BrokerHost mBrokerHost = getBrokerHost();
+    protected ITestBroker mBroker = getBroker();
+    @Nullable
+    protected abstract IAppInstaller brokerInstallationSource();
+    @Nullable
+    protected abstract String brokerHostApkName();
+
+    @NonNull
+    @Override
+    public BrokerHost getBrokerHost() {
+        // only initialize once....so calling getBrokerHost from anywhere returns the same instance
+        if (mBrokerHost == null) {
+            final String brokerHostApkName = brokerHostApkName();
+            mBrokerHost = new BrokerHost(brokerHostApkName);
+        }
+        return mBrokerHost;
+    }
+
+    @NonNull
+    @Override
+    public ITestBroker getBroker() {
+        // only initialize once....so calling getBroker from anywhere returns the same instance
+        if (mBroker == null) {
+            final IAppInstaller mBrokerSource = brokerInstallationSource();
+            final SupportedBrokers supportedBrokersAnnotation = getClass().getAnnotation(SupportedBrokers.class);
+            mBroker = BrokerTestHelper.createBrokerFromFlavor(supportedBrokersAnnotation, mBrokerSource);
+        }
+        return mBroker;
+    }
+
+    /**
+     * Run LabUserQuery to get a cloud device administrator user
+     *
+     * @return cloud device administrator user
+     */
+    public String getLabUserForCloudDeviceAdmin() {
+        final LabUserQuery query = new LabUserQuery();
+        query.userRole = LabConstants.UserRole.CLOUD_DEVICE_ADMINISTRATOR;
+        return LabUserHelper.loadUserForTest(query);
```

Comment:
you added the new lab api utilities but still using the old one. Why?

Reply:
I was using the new API for the jwtParser only, now that you mention I will change it